### PR TITLE
Upgrade gradle and android gradle plugin #964

### PR DESCRIPTION
**Pull request by wseemann**
Task/Issue URL: https://github.com/duckduckgo/Android/issues/964
Tech Design URL: N/A
CC: N/A

**Description**:
Upgrade gradle and android gradle plugin

**Steps to test this PR**:
1. Import app into Android Studio and build

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by wseemann** [comment]
@cmonfortep Is this PR OK to merge or should I close it?

**Comment by cmonfortep** [comment]
@wseemann thanks for asking. Since this PR requires AS 4.1, and it's not following our [contribution guidelines](https://github.com/duckduckgo/Android/blob/develop/CONTRIBUTING.md#contribute-code), we will discuss it internally. 

We might do an exception here since the contribution is not a big change, and we want to contribute to the Hacktoberfest, but first, we need to agree on moving everyone moving to AS 4.1.

**Comment by wseemann** [comment]
Thanks for the reply, as for the PR, can you explain to me what part doesn't fit the contribution guidelines? It was submitted no differently than the other PR I made so I'm confused. Thanks in advance.

**Comment by cmonfortep** [comment]
No problem @wseemann.

> "We welcome pull requests aimed at fixing bugs and security issues. We have also labeled tasks you can help with as help wanted."

TLDR; we expect only to get PRs from open issues that have been triaged by us. Our goal is to avoid people working on contributions that may not be finally merged due to incompatibilities with our roadmap or team setup. In this concrete case, the related issue was not triaged (see label "needs triage").

The other PR you mentioned was 100% fine, the related issue had the label "hacktoberfest", meaning that we accept any external contribution for Hacktoberfest.

P.S: if something was not clear enough in our contribution guidelines, please share it with us so I can improve them. Thanks!!

**Comment by wseemann** [comment]
That's for the reply. I now understand why this did not meet the guidelines. It'd probably be worthwhile to include a snippet in the contribution guidelines that states "please do not work on issues that have no been actively triaged first". Also, it's worth mentioning per the Hacktoberfest guidelines that opting-in/participating only requires the label on the repo itself, not the issue(s) within:

"We’ve changed the program to only count pull requests that are **made to repositories with a hacktoberfest topic or labeled with hacktoberfest-accepted**. Due to this change, it’s no longer necessary for maintainers to opt out of Hacktoberfest."

**Comment by cmonfortep** [comment]
@wseemann we discussed this internally, and we agreed on accepting this contribution. I will merge it once everyone has updated to AS 4.1. Thanks.

**Review by cmonfortep — APPROVED** [review]
